core/src/commonMain/kotlin/com/ustadmobile/core/controller/SchoolDetailOverviewPresenter.kt

```diff
@@ -0,0 +1,69 @@
+package com.ustadmobile.core.controller
+
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.impl.UmAccountManager
+import com.ustadmobile.core.impl.UstadMobileSystemImpl
+import com.ustadmobile.core.view.SchoolDetailOverviewView
+import com.ustadmobile.core.view.SchoolEditView
+import com.ustadmobile.door.DoorLifecycleOwner
+import com.ustadmobile.door.DoorLiveData
+import com.ustadmobile.lib.db.entities.SchoolWithHolidayCalendar
+
+import com.ustadmobile.lib.db.entities.UmAccount
+import com.ustadmobile.core.view.UstadView.Companion.ARG_ENTITY_UID
+import com.ustadmobile.lib.db.entities.School
+import kotlinx.coroutines.withTimeoutOrNull
+
+
+class SchoolDetailOverviewPresenter(context: Any,
+                                    arguments: Map<String, String>, view: SchoolDetailOverviewView,
+                                    lifecycleOwner: DoorLifecycleOwner,
+                                    systemImpl: UstadMobileSystemImpl,
+                                    db: UmAppDatabase, repo: UmAppDatabase,
+                                    activeAccount: DoorLiveData<UmAccount?> = UmAccountManager.activeAccountLiveData)
+    : UstadDetailPresenter<SchoolDetailOverviewView, SchoolWithHolidayCalendar>(context, arguments, view, lifecycleOwner, systemImpl,
+        db, repo, activeAccount) {
+
+    override val persistenceMode: PersistenceMode
+        get() = PersistenceMode.DB
+
+
+    override fun onCreate(savedState: Map<String, String>?) {
+        super.onCreate(savedState)
+    }
+
+    override suspend fun onLoadEntityFromDb(db: UmAppDatabase): SchoolWithHolidayCalendar? {
+        val entityUid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0L
+
+         val schoolWithHolidayCalendar = withTimeoutOrNull(2000) {
+             db.schoolDao.findByUidWithHolidayCalendarAsync(entityUid)
+         } ?: SchoolWithHolidayCalendar()
+
+        if(entityUid != 0L) {
```

Comment:
I think we don't need this if statement. entityUid for the detail presenter should never be 0. This presenter is loading from the database.